_posts/admin/2020-06-03-monitoring.md

```diff
@@ -53,6 +49,109 @@ The [project's documentation](https://bigbluebutton-exporter.greenstatic.dev/) i
+
+**Note:** You need to provide appropriate firewalling to those endpoints.
+
+### WebRTC SFU
+BigBlueButtons *Selective Forwarding Unit* can be configured to expose a number of Prometheus endpoints by adding the following snippets to
+`/etc/bigbluebutton/bbb-webrtc-sfu/production.yml`:
+
+```yaml
+prometheus:
+  enabled: true
+  # mcs-core specific metrics
+  host: localhost
+  port: 3014
+  path: '/metrics'
+  collectDefaultMetrics: false
+  # Main process metrics endpoint (main == websocket entrypoint, module manager)
+  main:
+    host: localhost
+    port: 3016
+    path: '/metrics'
+    collectDefaultMetrics: false
+  # Video process metrics endpoint (video == webcam req handler, ...)
+  video:
+    host: localhost
+    port: 3026
+    path: '/metrics'
+    collectDefaultMetrics: false
+  screenshare:
+    host: localhost
+    port: 3022
+    path: '/metrics'
+    collectDefaultMetrics: false
+  audio:
+    host: localhost
+    port: 3024
+    path: '/metrics'
+    collectDefaultMetrics: false
+```
+After a restart of bbb-webrtc-sfu, the metrics can now be queried via
+
+  * `curl http://bbb.example.com:3014/metrics` (mcs-core)
+  * `curl http://bbb.example.com:3016/metrics` (main)
+  * `curl http://bbb.example.com:3026/metrics` (video)
+  * `curl http://bbb.example.com:3022/metrics` (screenshare)
+  * `curl http://bbb.example.com:3024/metrics` (audio)
+
+respectively.
+
+### Meteor / HTML5 Frontend (experimental)
+The HTML5 Frontend servers can be configured to provide Prometheus endpoints by adding
+the following to `/etc/bigbluebutton/bbb-html5.yml`:
+
+```yaml
+private:
+    prometheus:
+        enabled: true
+        # Metrics endpoint path
+        path: '/metrics'
+        # Whether default metrics for Node.js processes should be exported
+        collectDefaultMetrics: false
+        # Whether redis metrics should be exported
+        collectRedisMetrics: false
+```
+
+After a restart, the metrics will be available at `https://bbb.example.com/html5client/metrics`.
+
+**Note:** The endpoint is unprotected. You need to configure nginx access control to this URL yourself.
+
+## Monitoring individual systemd services
+
+The third-party [systemd
+exporter](https://github.com/prometheus-community/systemd_exporter) can be used
+to monitor BigBlueButton services on a systemd level, i. e. track the usage of
+resources for every service. This allows fine-grained insights into the
+resource usage and distribution among the various components of BigBlueButton.
+
+The following command line will start the systemd exporter and instruct it to
+only monitor services relevant to BigBlueButton on port 9558:
+
+```shell
+systemd_exporter --systemd.collector.unit-include=(bbb.*|b3scale.*|freeswitch)\.service
+```
+It can be queried via `curl http://bbb.example.com:9558/metrics`.
+
+**Note:** You need to provide appropriate firewalling to those endpoints.
+
+## Optimization: Consolidating local metrics endpoints
```

Comment:
Add reference to this project as well: https://github.com/wrouesnel/reverse_exporter

Reply:
Do you know if there is there a benefit over the other?

Reply:
The only experience I have is with reverse_exporter, and it works well. My suggestion is to add it along with exporter_exporter so others will know there are alternatives.